assets/src/components/detoursTable.tsx

```diff
@@ -1,49 +1,107 @@
+    <tbody>
+      <DetourRows data={data} status={status} />
+    </tbody>
+  </Table>
+)
+
+const DetourRows = ({
+  data,
+  status,
+}: {
+  data: SimpleDetour[] | null
+  status: DetoursTableStatus
+}) =>
+  data === null ? (
+    <EmptyDetourRows message={`No ${status} detours.`} />
+  ) : (
+    <PopulatedDetourRows data={data} />
+  )
+
+const PopulatedDetourRows = ({ data }: { data: SimpleDetour[] }) => {
   const epochNowInSeconds = useCurrentTimeSeconds()
 
   return (
-    <Table hover className="c-detours-table">
-      <thead className="u-hide-for-mobile">
-        <tr>
-          <th className="px-3 py-4">Route and direction</th>
-          <th className="px-3 py-4 u-hide-for-mobile">Starting Intersection</th>
-          <th className="px-3 py-4 u-hide-for-mobile">On detour since</th>
-        </tr>
-      </thead>
-      <tbody>
-        {data.map((detour, index) => (
-          <tr key={index}>
-            <td className="align-middle p-3">
-              <div className="d-flex">
-                <RoutePill routeName={detour.route} />
-                <div className="c-detours-table__route-info-text d-inline-block">
-                  <div className="pb-1 fs-4 fw-semibold">{detour.name}</div>
-                  <div className="c-detours-table__route-info-direction fs-6">
-                    {detour.direction}
-                  </div>
+    <>
+      {data.map((detour, index) => (
+        <tr key={index}>
+          <td className="align-middle p-3">
+            <div className="d-flex">
+              <RoutePill routeName={detour.route} />
+              <div className="c-detours-table__route-info-text d-inline-block">
+                <div className="pb-1 fs-4 fw-semibold">{detour.name}</div>
+                <div className="c-detours-table__route-info-direction fs-6">
+                  {detour.direction}
                 </div>
               </div>
-            </td>
-            <td className="align-middle p-3 u-hide-for-mobile">
-              {detour.intersection}
-            </td>
-            <td className="align-middle p-3 u-hide-for-mobile">
-              {timeAgoLabel(epochNowInSeconds, detour.updatedAt)}
-            </td>
-          </tr>
-        ))}
-      </tbody>
-    </Table>
+            </div>
+          </td>
+          <td className="align-middle p-3 u-hide-for-mobile">
+            {detour.intersection}
+          </td>
+          <td className="align-middle p-3 u-hide-for-mobile">
+            {timeAgoLabel(epochNowInSeconds, detour.updatedAt)}
+          </td>
+        </tr>
+      ))}
+    </>
   )
 }
+
+const EmptyDetourRows = ({ message }: { message: string }) => (
+  <tr>
+    <td colSpan={3} className="p-3 p-md-4">
+      <div className="d-flex justify-content-center mb-3">
+        <EmptyDetourTableIcon height="100px" width="100px" />
+      </div>
+      <div className="d-flex justify-content-center">
+        <p className="fs-3 fw-light m-0">{message}</p>
+      </div>
+    </td>
+  </tr>
+)
```

Comment:
this comment is on the `<EmptyDetourRows/>` component
> issue(non-blocking): technically this is incorrect semantics and I think would readback strangely to a screen reader, but if you're okay with that we can keep it till it's an issue.
> https://github.com/mbta/skate/pull/2777#discussion_r1755235716

Reply:
I'd prefer to address this as a follow-up. No need to wait for it to become a problem, but I would like to get this merged so that it can unblock the things that come after.

Would you be up for pairing on this next time we're both in-office, since I'm still not totally sure what the problem is.

Reply:
> Would you be up for pairing on this

Yeah sure

Reply:
For posterity, the core of it is that to a screen reader (or really just the accessibility tree), this would read as "a table with 1 row" but the row just says (paraphrasing)"this table is empty"

Similar to 
![image](https://github.com/user-attachments/assets/6c417125-b5d0-40e2-befb-eb88a23eeb8b)

Reply:
@firestack would the right response here be to make the table headers (or even the entire table) be hidden to a screen reader, and replaced with just the text "No <status> detours"?

...with the `aria-hidden` attribute?

Reply:
I would keep the table and headers, because otherwise it would read as "Heading Detours, Heading Drafts"

and I think we want "Heading Detours, Table, 0 Rows"

You could either use the `<caption>` element, or add the content as it's own item outside of the table. But right now, I think it'd read as "Table, 1 row" and the row would say "There are no detours"(paraphrasing)

Reply:
Adding `aria-hidden` to the `EmptyDetourRows` makes that row disappear from the accessibility tree. So it wouldn't read the row saying "There are no detours" - it would just say "0 rows" or something like that.

But would it make more sense to get rid of the whole table, and _only_ have it say "No active detours"? So that it's something like "Heading Active Detours, No Active Detours"?

![Screenshot 2024-09-23 at 1 52 17 PM](https://github.com/user-attachments/assets/f5e7ed69-94fa-4a95-a1b4-ae22d1058376)